acquire/collector.py

```diff
@@ -372,21 +414,16 @@ def collect_path(
             log.error("- Failed to collect path %s", path, exc_info=True)
             return
 
-        if is_dir:
-            self.collect_dir(path, seen_paths=seen_paths, module_name=module_name)
+        if is_symlink and not follow:
+            self.collect_symlink(path, module_name=module_name, follow=follow, volatile=volatile)
+        elif is_dir:
+            self.collect_dir(path, seen_paths=seen_paths, module_name=module_name, follow=follow, volatile=volatile)
         elif is_file:
-            self.collect_file(path, module_name=module_name)
-        elif is_symlink:
-            self.report.add_path_failed(module_name, path)
-            log.error(
-                "- Can't collect %s (symlink to %s) in module %s",
-                path,
-                path.get().readlink(),
-                module_name,
-            )
```

Comment:
This error can still happen when there's a symlink that can't be resolved. In that case I think you'd also want to colelct the symlink itself (but still log an error/warning).

Reply:
I believe this code-path triggered when the entry is a symlink. Which causes no symlinks to be collected. I will add some error checking in the `collect_symlink()`